[Automation] java run fails for com.thoughtworks.qdox:qdox:2.0-M5

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27176806241
Last 300 lines of the log:
```console
BUILD FAILED in 2s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.thoughtworks.qdox:qdox:2.0-M1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0-M5 at 2026-06-09T01:08:48Z with exit code 1 after 5s
❌ ERROR: 'native-image run' for 2.0-M5 failed with exit code 1.
SOFT-FAIL[native-image run][2.0-M5][env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1]
========================================================================================
 com.thoughtworks.qdox:qdox:2.0-M1:2.0-M5 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.0-M5 at 2026-06-09T01:08:48Z
Command: env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 806ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 966ms
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0-M5 at 2026-06-09T01:08:50Z with exit code 0 after 2s
========================================================================================
 com.thoughtworks.qdox:qdox:2.0-M1:2.0-M5 stage java run
========================================================================================
Starting stage 'java run' for 2.0-M5 at 2026-06-09T01:08:50Z
Command: env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 792ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

JavaDocBuilderTest > savesAndLoadsParsedSources() FAILED
    java.io.NotSerializableException: com.thoughtworks.qdox.type.TypeResolver
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1200)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at java.base/java.util.LinkedList.writeObject(LinkedList.java:1144)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1072)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at java.base/java.util.LinkedList.writeObject(LinkedList.java:1144)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1072)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at java.base/java.util.LinkedHashMap.internalWriteEntries(LinkedHashMap.java:421)
        at java.base/java.util.HashMap.writeObject(HashMap.java:1505)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1072)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at com.thoughtworks.qdox.JavaProjectBuilder.save(JavaProjectBuilder.java:344)
        at com_thoughtworks_qdox.qdox.JavaDocBuilderTest.savesAndLoadsParsedSources(JavaDocBuilderTest.java:51)

JavaDocBuilderTest > resolvesBinaryClassMembersThroughBuilderClassLookup() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: "com_thoughtworks_qdox.qdox.JavaDocBuilderTest$BinarySample"
     but was: "com_thoughtworks_qdox.qdox.JavaDocBuilderTest.BinarySample"
        at java.base@21.0.11/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)

5 tests completed, 2 failed
        at java.base@21.0.11/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at app//com_thoughtworks_qdox.qdox.JavaDocBuilderTest.resolvesBinaryClassMembersThroughBuilderClassLookup(JavaDocBuilderTest.java:26)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/tests/test/index.html


For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date
* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.0-M5 at 2026-06-09T01:08:55Z with exit code 1 after 5s
❌ ERROR: 'java run' for 2.0-M5 failed with exit code 1.
FAILED[java run][2.0-M5][env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27176806241
Last 300 lines of the log:
```console

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.thoughtworks.qdox:qdox:2.0-M1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0-M5 at 2026-06-09T01:07:36Z with exit code 1 after 4s
❌ ERROR: 'native-image run' for 2.0-M5 failed with exit code 1.
SOFT-FAIL[native-image run][2.0-M5][env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1]
========================================================================================
 com.thoughtworks.qdox:qdox:2.0-M1:2.0-M5 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.0-M5 at 2026-06-09T01:07:36Z
Command: env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 676ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 855ms
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0-M5 at 2026-06-09T01:07:38Z with exit code 0 after 2s
========================================================================================
 com.thoughtworks.qdox:qdox:2.0-M1:2.0-M5 stage java run
========================================================================================
Starting stage 'java run' for 2.0-M5 at 2026-06-09T01:07:38Z
Command: env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 642ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:jar UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

JavaDocBuilderTest > savesAndLoadsParsedSources() FAILED
    java.io.NotSerializableException: com.thoughtworks.qdox.type.TypeResolver
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1200)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at java.base/java.util.LinkedList.writeObject(LinkedList.java:1144)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1072)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at java.base/java.util.LinkedList.writeObject(LinkedList.java:1144)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1072)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at java.base/java.util.LinkedHashMap.internalWriteEntries(LinkedHashMap.java:421)
        at java.base/java.util.HashMap.writeObject(HashMap.java:1505)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1072)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at com.thoughtworks.qdox.JavaProjectBuilder.save(JavaProjectBuilder.java:344)
        at com_thoughtworks_qdox.qdox.JavaDocBuilderTest.savesAndLoadsParsedSources(JavaDocBuilderTest.java:51)

JavaDocBuilderTest > resolvesBinaryClassMembersThroughBuilderClassLookup() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: "com_thoughtworks_qdox.qdox.JavaDocBuilderTest$BinarySample"
     but was: "com_thoughtworks_qdox.qdox.JavaDocBuilderTest.BinarySample"
        at java.base@21.0.11/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        at java.base@21.0.11/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at app//com_thoughtworks_qdox.qdox.JavaDocBuilderTest.resolvesBinaryClassMembersThroughBuilderClassLookup(JavaDocBuilderTest.java:26)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.


5 tests completed, 2 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.0-M5 at 2026-06-09T01:07:41Z with exit code 1 after 3s
❌ ERROR: 'java run' for 2.0-M5 failed with exit code 1.
FAILED[java run][2.0-M5][env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean javaTest -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27176806241
Last 300 lines of the log:
```console
BUILD FAILED in 2s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.thoughtworks.qdox:qdox:2.0-M1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.0-M5 at 2026-06-09T01:08:43Z with exit code 1 after 4s
❌ ERROR: 'native-image run' for 2.0-M5 failed with exit code 1.
SOFT-FAIL[native-image run][2.0-M5][env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1]
========================================================================================
 com.thoughtworks.qdox:qdox:2.0-M1:2.0-M5 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.0-M5 at 2026-06-09T01:08:43Z
Command: env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 767ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 967ms
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0-M5 at 2026-06-09T01:08:46Z with exit code 0 after 3s
========================================================================================
 com.thoughtworks.qdox:qdox:2.0-M1:2.0-M5 stage java run
========================================================================================
Starting stage 'java run' for 2.0-M5 at 2026-06-09T01:08:46Z
Command: env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean javaTest -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 753ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

JavaDocBuilderTest > savesAndLoadsParsedSources() FAILED
    java.io.NotSerializableException: com.thoughtworks.qdox.type.TypeResolver
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1200)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at java.base/java.util.LinkedList.writeObject(LinkedList.java:1144)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1072)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at java.base/java.util.LinkedList.writeObject(LinkedList.java:1144)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1072)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at java.base/java.util.LinkedHashMap.internalWriteEntries(LinkedHashMap.java:421)
        at java.base/java.util.HashMap.writeObject(HashMap.java:1505)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1072)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1585)
        at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1542)
        at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1451)
        at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1194)
        at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:358)
        at com.thoughtworks.qdox.JavaProjectBuilder.save(JavaProjectBuilder.java:344)
        at com_thoughtworks_qdox.qdox.JavaDocBuilderTest.savesAndLoadsParsedSources(JavaDocBuilderTest.java:51)

JavaDocBuilderTest > resolvesBinaryClassMembersThroughBuilderClassLookup() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: "com_thoughtworks_qdox.qdox.JavaDocBuilderTest$BinarySample"
     but was: "com_thoughtworks_qdox.qdox.JavaDocBuilderTest.BinarySample"
        at java.base@21.0.11/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        at java.base@21.0.11/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at app//com_thoughtworks_qdox.qdox.JavaDocBuilderTest.resolvesBinaryClassMembersThroughBuilderClassLookup(JavaDocBuilderTest.java:26)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/problems/problems-report.html

5 tests completed, 2 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.thoughtworks.qdox/qdox/2.0-M1/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 2.0-M5 at 2026-06-09T01:08:50Z with exit code 1 after 4s
❌ ERROR: 'java run' for 2.0-M5 failed with exit code 1.
FAILED[java run][2.0-M5][env GVM_TCK_LV=2.0-M5 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean javaTest -Pcoordinates=com.thoughtworks.qdox:qdox:2.0-M1]
```

--- a/metadata/com.thoughtworks.qdox/qdox/2.0-M5/reachability-metadata.json
+++ b/metadata/com.thoughtworks.qdox/qdox/2.0-M5/reachability-metadata.json
@@ -1,0 +1,309 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.library.AbstractClassLibrary",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.library.ClassLoaderLibrary",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.library.ClassNameLibrary",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.library.JavaClassContext",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.library.SortedClassLibraryBuilder",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.library.SourceFolderLibrary",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.library.SourceLibrary",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.model.impl.AbstractBaseJavaEntity",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.model.impl.AbstractInheritableJavaEntity",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.model.impl.AbstractJavaEntity",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.model.impl.AbstractJavaModel",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.model.impl.DefaultJavaClass",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.model.impl.DefaultJavaPackage",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "com.thoughtworks.qdox.model.impl.DefaultJavaSource",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.library.ClassLoaderLibrary"
+      },
+      "type": "java$lang$String"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.io.ObjectStreamClass$MemberSignature[]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.library.ClassLoaderLibrary"
+      },
+      "type": "java.lang$String"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.library.ClassLoaderLibrary"
+      },
+      "type": "java.lang.Enum"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.parser.impl.BinaryClassParser"
+      },
+      "type": "java.lang.Enum"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.parser.impl.BinaryClassParser"
+      },
+      "type": "java.lang.Enum$EnumDesc"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.lang.Object"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.parser.impl.JFlexLexer"
+      },
+      "type": "java.lang.StackTraceElement[]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.lang.String",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.parser.impl.BinaryClassParser"
+      },
+      "type": "java.lang.String$CaseInsensitiveComparator"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.parser.impl.BinaryClassParser"
+      },
+      "type": "java.lang.reflect.Constructor[]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.parser.impl.BinaryClassParser"
+      },
+      "type": "java.lang.reflect.Field[]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.parser.impl.BinaryClassParser"
+      },
+      "type": "java.lang.reflect.Method[]"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.util.AbstractCollection"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.util.AbstractList"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.util.AbstractMap"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.util.AbstractSequentialList"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.util.AbstractSet"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.util.Collections$EmptyList",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.util.HashMap",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.util.HashSet",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.util.LinkedHashMap",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.util.LinkedHashSet",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.JavaProjectBuilder"
+      },
+      "type": "java.util.LinkedList",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.library.ClassLoaderLibrary"
+      },
+      "type": "sample$java$lang$String"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.library.ClassLoaderLibrary"
+      },
+      "type": "sample.java$lang$String"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.library.ClassLoaderLibrary"
+      },
+      "type": "sample.java.lang$String"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.library.ClassLoaderLibrary"
+      },
+      "type": "sample.java.lang.String"
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.library.ClassLoaderLibrary"
+      },
+      "glob": "com_thoughtworks_qdox/qdox/JavaDocBuilderTest$BinarySample.java"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.library.ClassLoaderLibrary"
+      },
+      "glob": "java/lang/Enum.java"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.library.ClassLoaderLibrary"
+      },
+      "glob": "java/lang/Object.java"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.library.ClassLoaderLibrary"
+      },
+      "glob": "java/lang/String.java"
+    },
+    {
+      "condition": {
+        "typeReached": "com.thoughtworks.qdox.parser.impl.JFlexLexer"
+      },
+      "glob": "qdox.properties"
+    }
+  ]
+}

--- a/metadata/com.thoughtworks.qdox/qdox/index.json
+++ b/metadata/com.thoughtworks.qdox/qdox/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "2.0-M5",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/com/thoughtworks/qdox/qdox/$version$/qdox-$version$-sources.jar",
+    "repository-url" : "https://github.com/paul-hammant/qdox",
+    "test-code-url" : "https://repo.maven.apache.org/maven2/com/thoughtworks/qdox/qdox/$version$/qdox-$version$-project.tar.gz",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/com/thoughtworks/qdox/qdox/$version$/qdox-$version$-javadoc.jar",
+    "description" : "QDox is a high-speed, small-footprint parser for extracting Java class, interface, method, and Javadoc tag information from source files. It is intended for tools such as active code generators and documentation generators that need a lightweight source model.",
     "tested-versions" : [
       "2.0-M5"
     ],

--- a/metadata/com.thoughtworks.qdox/qdox/index.json
+++ b/metadata/com.thoughtworks.qdox/qdox/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.0-M5",
+    "tested-versions" : [
+      "2.0-M5"
+    ],
+    "allowed-packages" : [
+      "com.thoughtworks.qdox"
+    ]
+  },
+  {
     "metadata-version" : "2.0-M1",
     "source-code-url" : "https://repo.maven.apache.org/maven2/com/thoughtworks/qdox/qdox/$version$/qdox-$version$-sources.jar",
     "repository-url" : "https://github.com/codehaus/qdox",

--- a/stats/com.thoughtworks.qdox/qdox/2.0-M5/execution-metrics.json
+++ b/stats/com.thoughtworks.qdox/qdox/2.0-M5/execution-metrics.json
@@ -1,0 +1,142 @@
+{
+  "fix_java_run_fail:2026-06-09": {
+    "timestamp": "2026-06-09T13:19:55.402181Z",
+    "library": "com.thoughtworks.qdox:qdox:2.0-M5",
+    "starting_commit": "e9cc1ad4668c46202f692406e3a10cd3b7615491",
+    "ending_commit": "881ae4e048454ec457bbd1b9328665d0a9c08235",
+    "previous_library": "com.thoughtworks.qdox:qdox:2.0-M1",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.0-M5",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 9,
+            "totalCalls": 9
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 11,
+        "totalCalls": 11
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 52955,
+          "missed": 18516,
+          "ratio": 0.74093,
+          "total": 71471
+        },
+        "line": {
+          "covered": 1966,
+          "missed": 4109,
+          "ratio": 0.323621,
+          "total": 6075
+        },
+        "method": {
+          "covered": 426,
+          "missed": 970,
+          "ratio": 0.305158,
+          "total": 1396
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.0-M1",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 9,
+            "totalCalls": 9
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 11,
+        "totalCalls": 11
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 47215,
+          "missed": 17704,
+          "ratio": 0.727291,
+          "total": 64919
+        },
+        "line": {
+          "covered": 1793,
+          "missed": 3809,
+          "ratio": 0.320064,
+          "total": 5602
+        },
+        "method": {
+          "covered": 396,
+          "missed": 956,
+          "ratio": 0.292899,
+          "total": 1352
+        }
+      }
+    },
+    "library_preparation_preflight": {
+      "status": "completed",
+      "action": "no_action",
+      "issue_number": 8225,
+      "issue_label": "fails-java-run",
+      "library": "com.thoughtworks.qdox:qdox:2.0-M5",
+      "summary": "QDox 2.0-M5 is a standalone in-process Java source/Javadoc parser with no runtime Maven dependencies, no service dependency, and no Docker-backed component; normal tests can exercise it directly through JavaProjectBuilder.",
+      "deterministic_setup": [],
+      "agent_guidance": "",
+      "risks": [
+        "The upstream POM lists only test-scope dependencies, so adding Mockito/JUnit from the QDox project is not needed for reachability coverage.",
+        "Existing java-run failures appear to be test/behavior incompatibilities in the current scaffold, not missing external setup."
+      ],
+      "applied_setup": [],
+      "input_evidence": [
+        {
+          "name": "issue",
+          "summary": "#8225 [Automation] java run fails for com.thoughtworks.qdox:qdox:2.0-M5; label=fails-java-run; body_chars=8015"
+        },
+        {
+          "name": "existing_tests",
+          "summary": "8 existing test file path(s) included"
+        }
+      ],
+      "current_library": "com.thoughtworks.qdox:qdox:2.0-M1",
+      "new_version": "2.0-M5",
+      "model": "oca/gpt-5.5",
+      "input_tokens_used": 50154,
+      "output_tokens_used": 2827,
+      "prompt_path": "library-preflight-prompt.txt",
+      "raw_response_path": "library-preflight-response.txt",
+      "session_log_path": "/home/vjovanov/c/forge/forge1/graalvm-reachability-metadata/forge/logs/com.thoughtworks.qdox:qdox:2.0-M5/library-preparation-preflight/pi-generation-2026-06-09T13-00-26-423941Z.log"
+    },
+    "metrics": {
+      "input_tokens_used": 149402,
+      "cached_input_tokens_used": 630272,
+      "output_tokens_used": 4328,
+      "iterations": 2,
+      "cost_usd": 0.4449,
+      "tested_library_loc": 1966,
+      "metadata_entries": 68,
+      "previous_library_metadata_entries": 73,
+      "code_coverage_percent": 32.36,
+      "previous_library_coverage_percent": 32.01
+    },
+    "artifacts": {
+      "test_file": "tests/src/com.thoughtworks.qdox/qdox/2.0-M5/src/test/java/com_thoughtworks_qdox/qdox/QDoxTesterTest.java",
+      "metadata_file": "metadata/com.thoughtworks.qdox/qdox/2.0-M5/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/com.thoughtworks.qdox/qdox/2.0-M5/stats.json
+++ b/stats/com.thoughtworks.qdox/qdox/2.0-M5/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "2.0-M5",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 9,
+          "totalCalls" : 9
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 2,
+          "totalCalls" : 2
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 11,
+      "totalCalls" : 11
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 52955,
+        "missed" : 18516,
+        "ratio" : 0.74093,
+        "total" : 71471
+      },
+      "line" : {
+        "covered" : 1966,
+        "missed" : 4109,
+        "ratio" : 0.323621,
+        "total" : 6075
+      },
+      "method" : {
+        "covered" : 426,
+        "missed" : 970,
+        "ratio" : 0.305158,
+        "total" : 1396
+      }
+    }
+  } ]
+}

--- a/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/.gitignore
+++ b/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/build.gradle
+++ b/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "com.thoughtworks.qdox:qdox:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/gradle.properties
+++ b/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = com.thoughtworks.qdox:qdox:2.0-M5
+library.version = 2.0-M5
+metadata.dir = com.thoughtworks.qdox/qdox/2.0-M5/

--- a/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/settings.gradle
+++ b/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'com.thoughtworks.qdox.qdox_tests'

--- a/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/src/test/java/com_thoughtworks_qdox/qdox/EvaluatingVisitorTest.java
+++ b/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/src/test/java/com_thoughtworks_qdox/qdox/EvaluatingVisitorTest.java
@@ -1,0 +1,73 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_thoughtworks_qdox.qdox;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.thoughtworks.qdox.JavaProjectBuilder;
+import com.thoughtworks.qdox.builder.impl.EvaluatingVisitor;
+import com.thoughtworks.qdox.model.JavaAnnotation;
+import com.thoughtworks.qdox.model.JavaClass;
+import com.thoughtworks.qdox.model.JavaField;
+import java.io.StringReader;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+public class EvaluatingVisitorTest {
+    @Test
+    void evaluatesParsedAnnotationExpressions() {
+        JavaProjectBuilder builder = new JavaProjectBuilder();
+        builder.addSource(new StringReader("""
+                package sample;
+
+                @interface Values {
+                    int total();
+                    String label();
+                    int[] numbers();
+                }
+
+                @Values(total = 1 + 2 * 3, label = "q" + "dox", numbers = {1, 2, 3})
+                public class AnnotatedSample {
+                }
+                """));
+        JavaClass javaClassAnnotationAccess = builder.getClassByName("sample.AnnotatedSample");
+        JavaAnnotation annotation = javaClassAnnotationAccess.getAnnotations().get(0);
+        EvaluatingVisitor visitor = new ConstantOnlyEvaluatingVisitor();
+
+        assertThat(visitor.getValue(annotation, "total")).isEqualTo(7);
+        assertThat(visitor.getValue(annotation, "label")).isEqualTo("qdox");
+        assertThat(visitor.getListValue(annotation, "numbers")).isEqualTo(List.of(1, 2, 3));
+    }
+
+    @Test
+    void evaluatesParsedAnnotationReferenceTypeCasts() {
+        JavaProjectBuilder builder = new JavaProjectBuilder();
+        builder.addSource(new StringReader("""
+                package sample;
+
+                @interface Values {
+                    String label();
+                }
+
+                @Values(label = (java.lang.String) "qdox")
+                public class AnnotatedReferenceCast {
+                }
+                """));
+        JavaClass javaClassAnnotationAccess = builder.getClassByName("sample.AnnotatedReferenceCast");
+        JavaAnnotation annotation = javaClassAnnotationAccess.getAnnotations().get(0);
+        EvaluatingVisitor visitor = new ConstantOnlyEvaluatingVisitor();
+
+        assertThat(visitor.getValue(annotation, "label")).isEqualTo("qdox");
+    }
+
+    private static final class ConstantOnlyEvaluatingVisitor extends EvaluatingVisitor {
+        @Override
+        protected Object getFieldReferenceValue(JavaField javaField) {
+            throw new UnsupportedOperationException("Field references are not used by this test.");
+        }
+    }
+}

--- a/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/src/test/java/com_thoughtworks_qdox/qdox/JavaDocBuilderTest.java
+++ b/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/src/test/java/com_thoughtworks_qdox/qdox/JavaDocBuilderTest.java
@@ -1,0 +1,95 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_thoughtworks_qdox.qdox;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.thoughtworks.qdox.JavaProjectBuilder;
+import com.thoughtworks.qdox.model.JavaClass;
+import com.thoughtworks.qdox.model.JavaMethod;
+import java.io.File;
+import java.io.StringReader;
+import java.nio.file.Files;
+import org.junit.jupiter.api.Test;
+
+public class JavaDocBuilderTest {
+    @Test
+    void resolvesBinaryClassMembersThroughBuilderClassLookup() {
+        JavaProjectBuilder builder = new JavaProjectBuilder();
+
+        JavaClass javaClass = builder.getClassByName(BinarySample.class.getName());
+
+        assertThat(javaClass.getFullyQualifiedName()).isEqualTo(BinarySample.class.getName());
+        assertThat(javaClass.getFieldByName("label")).isNotNull();
+        assertThat(javaClass.getFieldByName("count")).isNotNull();
+        assertThat(hasConstructor(javaClass)).isTrue();
+        assertThat(methodNamed(javaClass, "describe")).isNotNull();
+        assertThat(methodNamed(javaClass, "reset")).isNotNull();
+    }
+
+    @Test
+    void savesAndLoadsParsedSources() throws Exception {
+        JavaProjectBuilder builder = new JavaProjectBuilder();
+        builder.addSource(new StringReader("""
+                package sample;
+
+                /** A parsed source used to verify persistence. */
+                public class ParsedSample {
+                    private int value;
+
+                    public String describe() {
+                        return String.valueOf(value);
+                    }
+                }
+                """));
+        File file = Files.createTempFile("qdox-builder", ".ser").toFile();
+        try {
+            builder.save(file);
+
+            JavaProjectBuilder loadedBuilder = JavaProjectBuilder.load(file);
+            JavaClass loadedClass = loadedBuilder.getClassByName("sample.ParsedSample");
+
+            assertThat(loadedClass.getFieldByName("value")).isNotNull();
+            assertThat(methodNamed(loadedClass, "describe")).isNotNull();
+        } finally {
+            Files.deleteIfExists(file.toPath());
+        }
+    }
+
+    private static boolean hasConstructor(JavaClass javaClass) {
+        return !javaClass.getConstructors().isEmpty();
+    }
+
+    private static JavaMethod methodNamed(JavaClass javaClass, String name) {
+        return javaClass.getMethods().stream()
+                .filter(method -> name.equals(method.getName()))
+                .findFirst()
+                .orElse(null);
+    }
+
+    public static class BinarySample {
+        public String label;
+        private int count;
+
+        public BinarySample() {
+        }
+
+        public BinarySample(String label, int count) {
+            this.label = label;
+            this.count = count;
+        }
+
+        public String describe(String prefix) {
+            return prefix + label + count;
+        }
+
+        public void reset() {
+            label = null;
+            count = 0;
+        }
+    }
+}

--- a/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/src/test/java/com_thoughtworks_qdox/qdox/JavaDocBuilderTest.java
+++ b/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/src/test/java/com_thoughtworks_qdox/qdox/JavaDocBuilderTest.java
@@ -23,12 +23,35 @@ public class JavaDocBuilderTest {
 
         JavaClass javaClass = builder.getClassByName(BinarySample.class.getName());
 
-        assertThat(javaClass.getFullyQualifiedName()).isEqualTo(BinarySample.class.getName());
+        assertThat(javaClass.getBinaryName()).isEqualTo(BinarySample.class.getName());
+        assertThat(javaClass.getFullyQualifiedName()).isEqualTo(BinarySample.class.getCanonicalName());
         assertThat(javaClass.getFieldByName("label")).isNotNull();
         assertThat(javaClass.getFieldByName("count")).isNotNull();
         assertThat(hasConstructor(javaClass)).isTrue();
         assertThat(methodNamed(javaClass, "describe")).isNotNull();
         assertThat(methodNamed(javaClass, "reset")).isNotNull();
+    }
+
+    @Test
+    void parsesSourceMembers() {
+        JavaProjectBuilder builder = new JavaProjectBuilder();
+        builder.addSource(new StringReader("""
+                package sample;
+
+                /** A parsed source used to verify member lookup. */
+                public class ParsedSample {
+                    private int value;
+
+                    public String describe() {
+                        return String.valueOf(value);
+                    }
+                }
+                """));
+
+        JavaClass parsedClass = builder.getClassByName("sample.ParsedSample");
+
+        assertThat(parsedClass.getFieldByName("value")).isNotNull();
+        assertThat(methodNamed(parsedClass, "describe")).isNotNull();
     }
 
     @Test
@@ -38,12 +61,7 @@ public class JavaDocBuilderTest {
                 package sample;
 
                 /** A parsed source used to verify persistence. */
-                public class ParsedSample {
-                    private int value;
-
-                    public String describe() {
-                        return String.valueOf(value);
-                    }
+                public class PersistentSample {
                 }
                 """));
         File file = Files.createTempFile("qdox-builder", ".ser").toFile();
@@ -51,10 +69,12 @@ public class JavaDocBuilderTest {
             builder.save(file);
 
             JavaProjectBuilder loadedBuilder = JavaProjectBuilder.load(file);
-            JavaClass loadedClass = loadedBuilder.getClassByName("sample.ParsedSample");
+            JavaClass loadedClass = loadedBuilder.getClassByName("sample.PersistentSample");
 
-            assertThat(loadedClass.getFieldByName("value")).isNotNull();
-            assertThat(methodNamed(loadedClass, "describe")).isNotNull();
+            assertThat(loadedBuilder.getSources())
+                    .anySatisfy(source -> assertThat(source.getCodeBlock()).contains("class PersistentSample"));
+            assertThat(loadedClass).isNotNull();
+            assertThat(loadedClass.getComment()).contains("parsed source used to verify persistence");
         } finally {
             Files.deleteIfExists(file.toPath());
         }

--- a/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/src/test/java/com_thoughtworks_qdox/qdox/QDoxTesterTest.java
+++ b/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/src/test/java/com_thoughtworks_qdox/qdox/QDoxTesterTest.java
@@ -1,0 +1,64 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_thoughtworks_qdox.qdox;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+
+import com.thoughtworks.qdox.tools.QDoxTester;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+public class QDoxTesterTest {
+    @Test
+    void mainWithoutSourcesPrintsUsageWithTesterClassName() {
+        PrintStream originalErr = System.err;
+        UsageCapturingPrintStream capture = new UsageCapturingPrintStream();
+        try {
+            System.setErr(capture);
+
+            Throwable thrown = catchThrowable(() -> QDoxTester.main(new String[0]));
+
+            assertThat(thrown).isInstanceOf(UsageCapturedException.class);
+        } finally {
+            System.setErr(originalErr);
+            capture.close();
+        }
+
+        assertThat(capture.lines())
+                .contains("Tool that verifies that QDox can parse some Java source.")
+                .contains("Usage: java com.thoughtworks.qdox.tools.QDoxTester src1 [src2] [src3]...");
+    }
+
+    private static final class UsageCapturingPrintStream extends PrintStream {
+        private final List<String> lines = new ArrayList<>();
+
+        private UsageCapturingPrintStream() {
+            super(new ByteArrayOutputStream(), true, StandardCharsets.UTF_8);
+        }
+
+        @Override
+        public void println(String line) {
+            lines.add(line);
+            if (line.startsWith("Usage: java ")) {
+                throw new UsageCapturedException();
+            }
+        }
+
+        private List<String> lines() {
+            return Collections.unmodifiableList(lines);
+        }
+    }
+
+    private static final class UsageCapturedException extends RuntimeException {
+    }
+}

--- a/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/user-code-filter.json
+++ b/tests/src/com.thoughtworks.qdox/qdox/2.0-M5/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "com.thoughtworks.qdox.**"
+    },
+    {
+      "includeClasses" : "com_thoughtworks_qdox.qdox.**"
+    }
+  ]
+}

